cmd/libsnap-confine-private/device-cgroup-support.c

```diff
@@ -330,6 +330,18 @@ static int _sc_cgroup_v2_init_bpf(sc_device_cgroup *self, int flags) {
     /* TODO: open("/sys/fs/bpf") and then mkdirat?  */
     /* create /sys/fs/bpf/snap as root */
     sc_identity old = sc_set_effective_identity(sc_root_group_identity());
+
+    /* we expect bpffs to be mounted at /sys/fs/bpf, which should have been done
+     * by systemd, but some systems out there are a weird mix of older userland
+     * and new kernels, in which case the assumptions about the state of the
+     * system no longer hold and we may need to mount bpffs ourselves */
+    if (!bpf_path_is_bpffs("/sys/fs/bpf")) {
+        debug("/sys/fs/bpf is not a bpffs mount");
+        /* bpffs isn't mounted at the usual place, try to do it now */
+        bpf_mount_bpffs("/sys/fs/bpf");
```

Comment:
Nitpick, "try" is a tiny bit confusing here since we `die` if it cannot be mounted. Maybe tweak the comment or make `bpf_mount_bpffs` return the status and die from here, like we do a few lines below when trying to create `/sys/fs/bpf/snap`?